cql3/statements/select_statement.cc

```diff
@@ -347,9 +347,10 @@ select_statement::do_execute(service::storage_proxy& proxy,
                                 auto timeout = db::timeout_clock::now() + timeout_duration;
                                 return p->fetch_page(builder, page_size, now, timeout);
                             }
-                    ).then([this, &builder, restrictions_need_filtering] {
+                    ).then([this, p, &builder, restrictions_need_filtering] {
```

Comment:
This makes a copy of p? What does that mean? Won't each copy have a separate copy of the statistics?

Reply:
`p` is a pointer - only one instance of the pager class (and thus its statistics) is used